### ISSUE TITLE
Adversarial Training for Black Box Attack

### ISSUE THREAD

**Issue by Aminullah6264** [Harry24k/adversarial-attacks-pytorch#54]
Could you please let me know, how can I update the Adversarial Training demo for the Black Box attack? 

https://github.com/Harry24k/adversarial-attacks-pytorch/blob/master/demos/Adversairal%20Training%20(MNIST).ipynb

**Comment by Harry24k** [comment]
Just change the model in the attack.
`model = #load holdout model`
`atk = PGD(model, eps=0.3, ...)`

**Comment by Aminullah6264** [comment]
Will a deep copy of the target model works as a holdout model?

`model = MyModel()
atk = PGD(model.deepcopy(), eps = ...`

**Comment by Harry24k** [comment]
Yes.With deepcopy, the model where adversarial examples are generated is
not changed.

2021년 12월 21일 (화) 오후 3:29, Amin Ullah ***@***.***>님이 작성:

**Comment by Amit-Nadam** [comment]
Hey,
is deepcopy() will work for all models and attacks in the library? I want to create a BlackBox version for all of them.

For example in this case, 
attack = torchattacks.CW(model.deepcopy(),..)
atk = attack(imgs,labels)

Thanks in advance,
Amit.
